firmware/powerboard_firmware/powerboard_firmware.ino

```diff
@@ -0,0 +1,125 @@
+size_t byteWritten = 0; // for incoming serial data
+
+// uint8_t all[11] = {0, 0, 0, 0, 0, 0, 0, 255, 25};
+uint8_t all[11] = {0, 0, 0, 0, 0, 0, 1, 255, 25};
+int voltage_pin = A0;
+
+typedef enum { 
+  GET_VOLTAGE = 0x22
+} CommandType;
+
+void setup() {
+  Serial.begin(9600); // opens serial port, sets data rate to 9600 bps
+}
+
+struct RequestType {
+  char command_id;
+  union {
+    // payload
+  };
+};
+
+struct ResponseType {
+  char command_id;
+  union {
+    int voltage;
+  };
+};
+
+void loop() {
+  // send data only when you receive data:
+  if (Serial.available() > 0) {
+    // read the incoming byte:
+    struct RequestType testing;
+    uint8_t encoded_request[11];
+    memcpy(encoded_request, &GET_VOLTAGE, sizeof(CommandType));
+    memcpy(encoded_request[9], &hash(encoded_request), sizeof(uint16_t));
+
+    testing = parse(encoded_request);
+    Serial.println(testing.command_id, DEC);
+
+    uint8_t *encoded_response = createResponse(testing);
+
+    // encoded_request[0] = GET_VOLTAGE;
+    
+
+    char buf[11];
+    byteWritten = Serial.readBytes(buf, 11);
```

Comment:
We should be reading one byte at a time and checking the hash every time.
Otherwise packets that aren't aligned to our read of 11 bytes will be missed.